openshift/machine-config-operator
# [MCO-126] Have capability to consume new format base OS image

An implementation of https://issues.redhat.com/browse/MCO-126. This should allow dual support for existing in-cluster OS image and new oci-format base os images.

To test, scale down CVO, then:

`oc -n openshift-machine-config-operator edit configmap/machine-config-osimageurl`

And switch to Colin's base image in https://github.com/openshift/os/pull/657#issuecomment-946926843 :

```osImageURL: registry.ci.openshift.org/coreos/walters-rhcos-ostreecontainer@sha256:3f57a0b046c023f837ae1c6d00f28e44a2a3c6201df556630698da29c942b2c8```

And the MCPs should update to something like:
```
# rpm-ostree status
State: idle
Deployments:
* ostree-unverified-registry:registry.ci.openshift.org/coreos/walters-rhcos-ostreecontainer@sha256:3f57a0b046c023f837ae1c6d00f28e44a2a3c6201df556630698da29c942b2c8
                    Digest: sha256:3f57a0b046c023f837ae1c6d00f28e44a2a3c6201df556630698da29c942b2c8
                   Version: 410.84.202201172236-0 (2022-01-17T22:38:34Z)

  pivot://registry.build01.ci.openshift.org/ci-ln-4zjjy02/stable@sha256:cd6b8a84e7609d99fcdc4c3005c73a1a7434a2610dc262f63b4497782f774b42
              CustomOrigin: Managed by machine-config-operator
                   Version: 410.84.202201201741-0 (2022-01-20T17:43:40Z)
```

Somewhat WIP status due to some uncertainty, comments inline
